controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -284,10 +305,14 @@ func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error)
 		clusterID := util.GetClusterID(r.ctx, r.Client, &r.Log)
 		if clusterID != "" && !strings.Contains(r.storageConsumer.Name, clusterID) {
 			if err := r.reconcileNoobaaAccount(); err != nil {
-				return reconcile.Result{}, err
+				return err
 			}
 		}
 
+		if err := r.deleteOnboardingSecret(); err != nil {
+			return nil
+		}
+
```

Comment:
This should go to the top to ensure we don't have a secret before creation of any resources

Reply:
done.